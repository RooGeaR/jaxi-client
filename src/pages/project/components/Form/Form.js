import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { Form, Field } from 'react-final-form';
import { userListQuery } from '../../../../queries';
import Styles from './Styles';

const ProjectForm = ({ title, buttonText, onSubmit, initialValues }) => (
	<Styles>
    <h1>{ title }</h1>
    <Form
      onSubmit={ onSubmit }
      initialValues={ initialValues }
      render={({ handleSubmit, form, submitting, pristine, values }) => (
        <form onSubmit={ handleSubmit }>
          <div>
            <label>Title</label>
            <Field
              name="title"
              component="input"
              type="text"
              placeholder="Title"
              required
            />
          </div>
          <div>
            <label>Responsible</label>
            <Field name="user_id" component="select" required>
              <Query query={ userListQuery } >
                {({ data, loading, error, fetchMore, refetch }) => {

                  if (loading) return <option> loading... </option>
                  if (error) return <option> { error.message } </option>
                  
                  return (
                    <>
                      {
                        data.users.map((user, i) => (
                          <option value={user.id} key={ i + 1 }>{ user.first_name } { user.last_name }</option>
                        ))
                      }
                    </>
                  );
                }}
              </Query>
            </Field>
          </div>
          <div className="buttons">
            <button type="submit" disabled={ submitting || pristine }>
              { buttonText }
            </button>
            <button
              type="button"
              onClick={ form.reset }
              disabled={ submitting || pristine }
            >
              Reset
            </button>
          </div>
          <pre>{JSON.stringify(values, 0, 2)}</pre>
        </form>
      )}
    />
  </Styles>
);

ProjectForm.propTypes = {
	/** Form title */
	title: PropTypes.string,

	/** Button submit text */
  buttonText: PropTypes.string,
  
  /** Function submit form */
  onSubmit: PropTypes.func,
  
  /** Initial values */
	initialValues: PropTypes.object,
};

export default ProjectForm;
