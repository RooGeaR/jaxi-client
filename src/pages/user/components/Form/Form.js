import React from 'react';
import PropTypes from 'prop-types';
import { Form, Field } from 'react-final-form';
import Styles from './Styles';

const UserForm = ({ title, buttonText, onSubmit, initialValues }) => (
	<Styles>
    <h1>{ title }</h1>
    <Form
      onSubmit={ onSubmit }
      initialValues={ initialValues }
      render={({ handleSubmit, form, submitting, pristine, values }) => (
        <form onSubmit={ handleSubmit }>
          <div>
            <label>First Name</label>
            <Field
              name="first_name"
              component="input"
              type="text"
              placeholder="First Name"
              required
            />
          </div>
          <div>
            <label>Last Name</label>
            <Field
              name="last_name"
              component="input"
              type="text"
              placeholder="Last Name"
              required
            />
          </div>
          <div>
            <label>Email</label>
            <Field
              name="email"
              component="input"
              type="text"
              placeholder="Email"
              required
            />
          </div>
          <div>
            <label>Username</label>
            <Field
              name="username"
              component="input"
              type="text"
              placeholder="Username"
              required
            />
          </div>
          <div>
            <label>Password</label>
            <Field
              name="password"
              component="input"
              type="password"
              placeholder="Password"
              required
            />
          </div>
          <div className="buttons">
            <button type="submit" disabled={ submitting || pristine } id='submit-button'>
              { buttonText }
            </button>
            <button
              type="button"
              onClick={ form.reset }
              disabled={ submitting || pristine }
            >
              Reset
            </button>
          </div>
        </form>
      )}
    />
  </Styles>
);

UserForm.propTypes = {
	/** Form title */
	title: PropTypes.string,

	/** Button submit text */
  buttonText: PropTypes.string,
  
  /** Function submit form */
  onSubmit: PropTypes.func,
  
  /** Initial values */
	initialValues: PropTypes.object,
};

export default UserForm;
