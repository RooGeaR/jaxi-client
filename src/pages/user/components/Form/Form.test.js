import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

const onSubmit = jest.fn();

describe('Create user form', () => {
  it('should render correctly', () => {
    const component = shallow(<Form title='Create user' buttonText='Save' onSubmit={onSubmit} />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('Update user form', () => {
  it('should render correctly', () => {
    const component = shallow(<Form title='Update user' buttonText='Update' onSubmit={onSubmit} />);
  
    expect(component).toMatchSnapshot();
  });
});
