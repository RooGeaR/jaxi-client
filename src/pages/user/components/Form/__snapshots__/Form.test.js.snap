// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create user form should render correctly 1`] = `
<styled.div>
  <h1>
    Create user
  </h1>
  <ReactFinalForm
    onSubmit={[MockFunction]}
    render={[Function]}
  />
</styled.div>
`;

exports[`Update user form should render correctly 1`] = `
<styled.div>
  <h1>
    Update user
  </h1>
  <ReactFinalForm
    onSubmit={[MockFunction]}
    render={[Function]}
  />
</styled.div>
`;
